cl/corpus_importer/management/commands/recap_into_opinions.py

```diff
@@ -0,0 +1,127 @@
+import eyecite
+from asgiref.sync import async_to_sync
+from django.core.management import BaseCommand
+from django.db import transaction
+
+from cl.lib.command_utils import logger
+from cl.lib.microservice_utils import microservice
+from cl.search.models import (
+    SOURCES,
+    Court,
+    Opinion,
+    OpinionCluster,
+    RECAPDocument,
+)
+
+
+def import_opinions_from_recap(court=None, total_count=0):
+    """Import recap documents into opinion db
+
+    :param court: Court ID if any
+    :param total_count: The number of new opinions to add
+    :return: None
+    """
+    count = 0
+    if not court:
+        courts = Court.objects.filter(
+            jurisdiction=Court.FEDERAL_DISTRICT
+        ).exclude(
+            id__in=["orld", "dcd"]
+        )  # orld is historical and we gather dcd opinions from the court
+    else:
+        courts = Court.objects.filter(pk=court)
+    for court in courts:
+        logger.info(f"Importing RECAP documents for {court}")
+        cluster = (
+            OpinionCluster.objects.filter(docket__court=court)
+            .exclude(source=SOURCES.RECAP)
+            .order_by("-date_filed")
+            .first()
+        )
+
+        documents = RECAPDocument.objects.filter(
+            docket_entry__docket__court=court,
+            docket_entry__docket__date_filed__gt=cluster.date_filed,
+            is_available=True,
+            is_free_on_pacer=True,
+        ).order_by("id")
+
+        for recap_document in documents.iterator():
+            docket = recap_document.docket_entry.docket
+            if "cv" not in docket.docket_number.lower():
+                logger.info(f"Skipping non civil opinion")
+                continue
+
+            ops = Opinion.objects.filter(sha1=recap_document.sha1)
+            if ops.count() > 0:
+                logger.info(
+                    f"Skipping previously imported opinion: {ops[0].id}"
+                )
+                continue
+
+            response = async_to_sync(microservice)(
+                service="recap-extract",
+                item=recap_document,
+                params={"strip_margin": True},
+            )
+
+            response.raise_for_status()
+            cites = eyecite.get_citations(response.json()["content"])
```

Comment:
I think you'll want to use `filter_out_non_case_law_citations`, and the HYPERSCAN_TOKENIZER, for speed.

Reply:
yep